wfe2/wfe.go

```diff
@@ -492,31 +493,40 @@ func (wfe *WebFrontEndImpl) NewAccount(
 		return
 	}
 
-	var init core.Registration
+	type accountCreateRequest struct {
+		Contact              *[]string `json:"contact"`
+		TermsOfServiceAgreed bool      `json:"termsOfServiceAgreed"`
+	}
+	var init accountCreateRequest
+
 	err = json.Unmarshal(body, &init)
 	if err != nil {
 		wfe.sendError(response, logEvent, probs.Malformed("Error unmarshaling JSON"), err)
 		return
 	}
-	if len(init.Agreement) > 0 && init.Agreement != wfe.SubscriberAgreementURL {
-		msg := fmt.Sprintf("Provided agreement URL [%s] does not match current agreement URL [%s]", init.Agreement, wfe.SubscriberAgreementURL)
-		wfe.sendError(response, logEvent, probs.Malformed(msg), nil)
+	if !init.TermsOfServiceAgreed {
+		wfe.sendError(response, logEvent, probs.Malformed("must agree to terms of service"), nil)
 		return
 	}
-	init.Key = key
-	init.InitialIP = net.ParseIP(request.Header.Get("X-Real-IP"))
-	if init.InitialIP == nil {
+
+	ip := net.ParseIP(request.Header.Get("X-Real-IP"))
+	if ip == nil {
 		host, _, err := net.SplitHostPort(request.RemoteAddr)
 		if err == nil {
-			init.InitialIP = net.ParseIP(host)
+			ip = net.ParseIP(host)
 		} else {
 			logEvent.AddError("Couldn't parse RemoteAddr: %s", request.RemoteAddr)
 			wfe.sendError(response, logEvent, probs.ServerInternal("couldn't parse the remote (that is, the client's) address"), nil)
 			return
 		}
 	}
 
-	acct, err := wfe.RA.NewRegistration(ctx, init)
+	acct, err := wfe.RA.NewRegistration(ctx, core.Registration{
```

Comment:
Thanks for changing this to an explicit `core.Registration{}` constructed on the fly instead of mutating the user provided struct. Much nicer! :nail_care: